feat: Offensive and Humor Input in AffectiveAnnotationProject

artikandri opened the pull request:
Features added with this PR: 
- Offensive Input will be shown when the AffectiveAnnotation project is_offensive_mode mode is set to `true`
- Summary Input will be shown when the AffectiveAnnotation project is_summary_mode mode is set to `true`
- Humor Input will be shown when the AffectiveAnnotation project is_humor_mode mode is set to `true`

What's changed: 
- `frontend/.eslintrc.js`: increased the limit of max characters in one line from 100 to 150
- `frontend/i18n/*/projects/annotation.js`: translation for the newly added project modes questions
- `frontend/i18n/*/rules.js`: required rules translation
- `frontend/pages/projects/_id/affective-annotation/index.vue`: implemented the new inputs; removed the blank space in the annotation box

What's new: 
- `frontend/components/tasks/affectiveAnnotation/humor/HumorInput.vue`: Humor Input component for  AffectiveAnnotation project
- `frontend/components/tasks/affectiveAnnotation/offensive/OffensiveInput.vue`: Offensive Input component for  AffectiveAnnotation project

Notable: 
- Now that I noticed it, the styling looks different from this PR: https://github.com/CLARIN-PL/doccano/pull/31  XD I think we can choose which style to be used and one of us will follow XD 


Screenshots: 
![image](https://user-images.githubusercontent.com/12537724/194825798-3ba21bb2-9251-4151-aea1-e0ba2a8a84ed.png)
![image](https://user-images.githubusercontent.com/12537724/194825833-3af6147f-d751-4d37-a252-6f3791f334b3.png)
![image](https://user-images.githubusercontent.com/12537724/194825913-820aca0f-3e55-4a04-b403-31aaf7b6292f.png)

teddy-f-47 commented on frontend/components/tasks/affectiveAnnotation/humor/HumorInput.vue:
```diff
@@ -0,0 +1,252 @@
+                                        outlined
+                                        :required="substatement.isChecked"
+                                        autofocus
+                                        :label="
+                                            $t(`annotation.humor.subquestion3.substatement${(idx+1)}Question`)"
+                                        v-model.trim="substatement.reason" 
+                                        :rules="[rules.required]"
+                                        class="subquestions-item__textfield" />
```
This is very minor, but I got a warning saying that `Attribute "v-model.trim" should go before ":label"  vue/attributes-order`, so maybe the order should be adjusted 😅

artikandri replied:
Okiee XD Thank you, I missed to check `yarn` tab before submitting  the PR.

teddy-f-47 commented on frontend/components/tasks/affectiveAnnotation/offensive/OffensiveInput.vue:
```diff
@@ -0,0 +1,240 @@
+                                        autofocus
+                                        :label="
+                                            $t(`annotation.offensive.subquestion3.substatement${(idx+1)}Question`)"
+                                        v-model.trim="substatement.reason" 
```
Similar case, `warning  Attribute "v-model.trim" should go before ":label"  vue/attributes-order` 😅

teddy-f-47 commented on frontend/pages/projects/_id/affective-annotation/index.vue:
```diff
@@ -89,6 +89,7 @@
                 />
               </div>
             </v-card>
+            <component :is="affectiveAnnotationComponent" v-if="affectiveAnnotationComponent" />
           </v-col>
```
I like this approach of importing the right input component based on the mode, it looks cleaner and more concise. I just have a question, how do we set the props values for a specific input component in this approach? XD

artikandri replied:
I have seen your code thoroughly, and with your approach, yes; it will be kind of hard to integrate them . The props can be set like usual actually, but it will not be clean and readable so I will discard this approach and follow yours.  Better than rewriting the whole code. XD

teddy-f-47 reviewed (commented):
Thank you for the great work! In general the code looks very good, I just have a few comments. XD

1. In offensiveness mode, some dimensions are missing (there are new changes in the requirements file):
 - (already present) Rasę, kolor skóry, wygląd [x] jaki? [textbox]
 - (missing) Wiek [x] jaki? [textbox]
 - (missing) Czyn, działanie [x] jakie? [textbox]
 - (already present) Orientację seksualną [x] jaką?  [textbox]
 - Poglądy polityczne, **partie** (an additional word in the prompt question)
2. Similarly, in the emotions mode, these new dimensions are also added.
3. About the validation mechanism related to the requirement: "if there is "0" on two sliders before, user does not need to click......." I think we don't need to implement it now. but later imo we can add it inside the "confirm as done" button here:
<img width="315" alt="ss22" src="https://user-images.githubusercontent.com/64476430/194863494-3d245f57-7b06-4e32-8738-23da438c3d25.PNG"> What do you think? XD
4. About the styling, imo this one is fine. I think slightly different styling in different modes can be beneficial in that it helps the user quickly/intuitively see what mode they are in, so for now I will keep the styling in Emotions and Others, and you can use the current styling for Offensiveness and Humor. But then again, we can discuss if we want to make everything the same. XD
5. Imo for the textfields, you may want to use the TextfieldWithSeq2Seq component, because it will be easier to handle when/how to send the inputted text to the BE later. With textfield alone, we will need to consider when to send the data to BE (whether on key up, or on enter, or something else). But this is just a suggestion, you may have a different idea and it is totally fine XD

Finally, don't forget to rebase and resolve the conflicts, because of the recent merge 😅

haanh764 reviewed (commented):
Thank you for such great work! Generally, the code looks fine but I have similar comments as Teddy also. So I will wait for the next changes XD

About (3) in Teddy's comment, imo I think if the 2 first sliders are 0, maybe we will disable all the next questions (or even not showing them). It will be more straightforward than using the "confirm as done" button since this button has different function. So you can consider XD

artikandri commented:
@teddy-f-47 
1. 2. Thank you for pointing it out! I didn't compare to the newest requirements XD I will add them
3. I think I will go with Anh's approach about that XD   Would be better not to modify the other component 
4. I think the mode can be shown with another approach, so I will follow yours perhaps, with a slight modification because I want to show the `Wcale` and `W pełni` texts. 
5. To be honest, I don't really like using the TextfieldWithSeq2Seq component because it was made to handle keywords and we are not handling keywords here, the UX is different  :<  I like the approach of using `modal` to show the whole component and that we can use the `Enter` button tho. So I will modify it a bit probably. 


@haanh764 
I like that suggestion, thank you XD  I will use it. Sorry, I missed that requirement.

artikandri commented:
Hello everyone, sorry for the long update XD  I previously thought that I will also add the API in this PR, so I didn't rerequest the review. 

Anyway, I have fixed the styling and added the new requirements. In addition to adjusting the slider styling and disabled/reduced the opacity when user hasnt filled the slider, I also adjusted the textfield so that it will open in a new modal and behave in the same way like Teddy's for consistencies.  I hope I didn't miss anything else now XD 

![image](https://user-images.githubusercontent.com/12537724/195601400-aeea5a4b-6b85-40eb-bfcc-884ead79b214.png)
![image](https://user-images.githubusercontent.com/12537724/195601609-fbb480f6-7a7b-4f95-92c3-48cd9aebf5b6.png)

 @haanh764  @teddy-f-47

teddy-f-47 reviewed (approved):
Hi, sorry for taking some time to response 😅 I have seen the code and I think everything is fine. I like that you refactored the `async fetch()` so that setDoc and setArticleData are separated. XD Great job, approved for merging!

Btw for the api connection, I will try to update my PR either today or tomorrow at the latest so that it can be merged, and then you can add api connections for these inputs 😅 I'm not sure tho, do you want to include the api connections in this PR or a separate PR?

artikandri commented:
> Hi, sorry for taking some time to response 😅 I have seen the code and I think everything is fine. I like that you refactored the `async fetch()` so that setDoc and setArticleData are separated. XD Great job, approved for merging!
> 
> Btw for the api connection, I will try to update my PR either today or tomorrow at the latest so that it can be merged, and then you can add api connections for these inputs 😅 I'm not sure tho, do you want to include the api connections in this PR or a separate PR?

I will include them in separate PR xD

haanh764 reviewed (approved):
Sorry for the late checking, I missed the update comments XD I think everything looks great now, thank you for your work! Approved!
